MastermindTests/ViewTests.cs

```diff
@@ -0,0 +1,121 @@
+            
+            _mockedInputOutput.Setup(output => output.OutputWelcomeMessage())
+                .Verifiable();
+        
+            // Act
+            _view.DisplayInitialMessage();
+        
+            // Assert
+            _mockedInputOutput.Verify();
+        }
+        
+        [Fact]
+        public void GivenAViewInstanceWithAConsoleDependency_WhenTheGetUpdatedUserGuessIsCalledAfterUpdatePlayerGuess_ThenShouldReturnAnArrayOfColoursContainingTheUserGuess()
+        {
+            // Arrange
+            var expectedGuess = new [] {Colour.Red, Colour.Blue, Colour.Yellow, Colour.Green};
+            var game = new Game();
+            
+            _mockedInputOutput.Setup(output => output.GetAGuessInput())
+                .Returns(new PlayerInput
+                { 
+                    ColoursInput = new[] {ColourSquares.Red, ColourSquares.Blue, ColourSquares.Yellow, ColourSquares.Green}
+                })
+                .Verifiable();
+        
+            // Act
+            _view.UpdateLastPlayerGuessInGame(game);
+            var actualGuess = game.LatestPlayerGuess;
+        
+            // Assert
+            _mockedInputOutput.Verify();
+            Assert.Equal(expectedGuess, actualGuess);
+        }
+        
+        [Fact]
+        public void GivenAViewInstanceWithAConsoleDependency_WhenTheDisplayGameInfoIsCalled_ThenShouldDisplayAStringWithAllTheClues()
+        {
+            // Arrange
+            var view = new View(_mockedInputOutput.Object);
+            var clueInput = new List<Clue> {Clue.Black, Clue.White, Clue.Black};
+            _mockedInputOutput.Setup(output => 
+                    output.OutputClues(new List<string> {ColourSquares.Black, ColourSquares.White, ColourSquares.Black}))
+                .Verifiable();
+        
+            // Act
+            view.DisplayGameInfo(new Game()
+            {
+                GameState = GameStatus.Playing,
+                Clues = clueInput,
+                GuessingCount = 59,
+                LatestPlayerGuess = new [] { Colour.Blue }
+            });
+        
+            // Assert
+            _mockedInputOutput.Verify();
+        }
+        
+        [Theory]
+        [InlineData(GameStatus.Won)]
+        [InlineData(GameStatus.Quit)]
+        public void GivenAViewInstanceWithAConsoleDependency_WhenDisplayEndGameResultIsCalled_ThenShouldDisplayTheAnswerWithAFinalGameMessage(GameStatus gameStatus)
+        {
+            // Arrange
+            var correctAnswer = new[] {Colour.Blue, Colour.Red, Colour.Yellow, Colour.Green};
+
+            if (gameStatus == GameStatus.Won)
+            {
+                _mockedInputOutput.Setup(output => 
+                    output.OutputGameWonMessage())
+                .Verifiable();
+            }
+            else
+            {
+                _mockedInputOutput.Setup(output => 
+                        output.OutputGameQuitMessage())
+                    .Verifiable();
+            }
```

Comment:
My personal preference would be to split this out into two separate tests. It might be a bit more verbose but it can help to avoid some potential complexity later down the road if other game statuses are introduced later down the road.